test/validation/api/ml/ml.c

```diff
@@ -0,0 +1,501 @@
+	CU_ASSERT(ml_capa.max_models_loaded > 0);
+	CU_ASSERT(ml_capa.max_inputs > 0);
+	CU_ASSERT(ml_capa.max_outputs > 0);
+	CU_ASSERT(ml_capa.max_compl_id > 0);
+	CU_ASSERT(ml_capa.max_segs_per_input > 0);
+	CU_ASSERT(ml_capa.max_segs_per_output > 0);
+	CU_ASSERT(ml_capa.min_input_align > 0);
+	CU_ASSERT(ml_capa.min_output_align > 0);
+
+	if ((ml_capa.load.compl_mode_mask | ml_capa.run.compl_mode_mask) &
+	    ODP_ML_COMPL_MODE_EVENT) {
+		odp_pool_capability_t pool_capa;
+
+		CU_ASSERT_FATAL(odp_pool_capability(&pool_capa) == 0);
+
+		CU_ASSERT(ml_capa.pool.max_pools > 0);
+		CU_ASSERT(ml_capa.pool.max_pools <= pool_capa.max_pools);
+		CU_ASSERT(ml_capa.pool.max_num > 0);
+		CU_ASSERT(ml_capa.pool.max_cache_size >= ml_capa.pool.min_cache_size);
+	}
+
+	CU_ASSERT(ml_capa.load.compl_mode_mask);
+	CU_ASSERT(ml_capa.run.compl_mode_mask);
+
+	if (ml_capa.load.compl_mode_mask & ODP_ML_COMPL_MODE_POLL ||
+	    ml_capa.run.compl_mode_mask & ODP_ML_COMPL_MODE_POLL)
+		CU_ASSERT(ml_capa.max_compl_id > 0);
+
+	if (ml_capa.load.compl_mode_mask & ODP_ML_COMPL_MODE_EVENT)
+		CU_ASSERT(ml_capa.load.compl_queue_plain || ml_capa.load.compl_queue_sched);
+
+	if (ml_capa.run.compl_mode_mask & ODP_ML_COMPL_MODE_EVENT)
+		CU_ASSERT(ml_capa.run.compl_queue_plain || ml_capa.run.compl_queue_sched);
+}
+
+static void test_ml_param(uint8_t fill)
+{
+	odp_ml_model_param_t model_param;
+	odp_ml_compl_pool_param_t pool_param;
+	odp_ml_compl_param_t compl_param;
+	odp_ml_run_param_t run_param;
+
+	memset(&model_param, fill, sizeof(model_param));
+	odp_ml_model_param_init(&model_param);
+	CU_ASSERT_EQUAL(model_param.max_compl_id, 0);
+	CU_ASSERT_FALSE(model_param.extra_stat_enable);
+	CU_ASSERT_PTR_NULL(model_param.extra_param);
+	CU_ASSERT_EQUAL(model_param.extra_info.num_inputs, 0);
+	CU_ASSERT_EQUAL(model_param.extra_info.num_outputs, 0);
+	CU_ASSERT_PTR_NULL(model_param.extra_info.input_format);
+	CU_ASSERT_PTR_NULL(model_param.extra_info.output_format);
+
+	memset(&pool_param, fill, sizeof(pool_param));
+	odp_ml_compl_pool_param_init(&pool_param);
+	CU_ASSERT_EQUAL(pool_param.uarea_size, 0);
+	CU_ASSERT_PTR_NULL(pool_param.uarea_init.args);
+	CU_ASSERT(pool_param.uarea_init.init_fn == NULL);
+	CU_ASSERT(pool_param.cache_size <= global.ml_capa.pool.max_cache_size);
+	CU_ASSERT(pool_param.cache_size >= global.ml_capa.pool.min_cache_size);
+
+	memset(&compl_param, fill, sizeof(compl_param));
+	odp_ml_compl_param_init(&compl_param);
+	CU_ASSERT_PTR_NULL(compl_param.user_ptr);
+
+	memset(&run_param, fill, sizeof(run_param));
+	odp_ml_run_param_init(&run_param);
+	CU_ASSERT_EQUAL(run_param.batch_size, 0);
+	CU_ASSERT_PTR_NULL(run_param.result);
+}
+
+static void test_ml_param_init(void)
+{
+	test_ml_param(0x00);
+	test_ml_param(0xff);
+}
+
+static void test_ml_debug(void)
+{
+	odp_ml_print();
+}
+
+static void ml_compl_pool_create_max_pools(void)
+{
+	int ret;
+	uint32_t i, j;
+	odp_ml_compl_pool_param_t ml_pool_param;
+	uint32_t max_pools = global.ml_capa.pool.max_pools - 1;
```

Comment:
Is there a reason to test only max_pools - 1? E.g. I didn't notice any pool created in the init phase. If there's reason, this line should have a comment about that - otherwise should test capa max_pools.